### ISSUE TITLE
average loss increase instead of decrease

### ISSUE THREAD

**Issue by aligoglos** [davisking/dlib#1065]
I want train new deep face recognition on [FEI ](http://fei.edu.br/~cet/facedatabase.html) data set.
i choose resnet model offer dlib:
```
// ----------------------------------------------------------------------------------------
//					Network Definition
//
template <template <int, template<typename>class, int, typename> class block, int N, template<typename>class BN, typename SUBNET>
using residual = add_prev1<block<N, BN, 1, tag1<SUBNET>>>;

template <template <int, template<typename>class, int, typename> class block, int N, template<typename>class BN, typename SUBNET>
using residual_down = add_prev2<avg_pool<2, 2, 2, 2, skip1<tag2<block<N, BN, 2, tag1<SUBNET>>>>>>;

template <int N, template <typename> class BN, int stride, typename SUBNET>
using block = BN<con<N, 3, 3, 1, 1, relu<BN<con<N, 3, 3, stride, stride, SUBNET>>>>>;


template <int N, typename SUBNET> using ares = relu<residual<block, N, affine, SUBNET>>;
template <int N, typename SUBNET> using ares_down = relu<residual_down<block, N, affine, SUBNET>>;

// ----------------------------------------------------------------------------------------

template <typename SUBNET> using alevel0 = ares_down<256, SUBNET>;
template <typename SUBNET> using alevel1 = ares<256, ares<256, ares_down<256, SUBNET>>>;
template <typename SUBNET> using alevel2 = ares<128, ares<128, ares_down<128, SUBNET>>>;
template <typename SUBNET> using alevel3 = ares<64, ares<64, ares<64, ares_down<64, SUBNET>>>>;
template <typename SUBNET> using alevel4 = ares<32, ares<32, ares<32, SUBNET>>>;

using anet_type = loss_metric<fc_no_bias<128, avg_pool_everything<
	alevel0<
	alevel1<
	alevel2<
	alevel3<
	alevel4<
	max_pool<3, 3, 2, 2, relu<affine<con<32, 7, 7, 2, 2,
	input_rgb_image_sized<150>
	>>>>>>>>>>>>;
```
and choose trainer like this :
```
anet_type net;
dnn_trainer<anet_type, adam> trainer(net, adam(0.0005, 0.9, 0.999));
trainer.set_min_learning_rate(0.000001);
trainer.set_mini_batch_size(128);
trainer.be_verbose();
trainer.set_learning_rate(0.0001);
trainer.set_synchronization_file("face_metric_sync", std::chrono::minutes(5));
trainer.set_iterations_without_progress_threshold(30000);
```
but average loss increase instead of decrease !? i test different learning rate but not differ or average loss become nan, :
```
Epoch: 1     learning rate: 0.0001  average loss: 0.779637     steps without apparent progress: 0
Epoch: 2     learning rate: 0.0001  average loss: 1.4496       steps without apparent progress: 0
Epoch: 3     learning rate: 0.0001  average loss: 3.52596      steps without apparent progress: 0
Epoch: 4     learning rate: 0.0001  average loss: 19.2634      steps without apparent progress: 0
Epoch: 5     learning rate: 0.0001  average loss: 138.815      steps without apparent progress: 0
Epoch: 6     learning rate: 0.0001  average loss: 1191.8       steps without apparent progress: 0
Epoch: 7     learning rate: 0.0001  average loss: 14556.1      steps without apparent progress: 0
Epoch: 8     learning rate: 0.0001  average loss: 221689       steps without apparent progress: 0
Epoch: 9     learning rate: 0.0001  average loss: 4.60275e+06  steps without apparent progress: 0
```
how to fixed that?
in my case i choose this image :
> *-01
> *-03
> *-05
> *-07
> *-09
> *-11
for training and other for test.

**Comment by davisking** [comment]
Fill out the issue template.
